1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_2_Palindrome/src/index.ts

```diff
@@ -1,11 +1,11 @@
 export function reverseAString(str: string) {
-    return str.toLowerCase().split("").reverse().join("");
+    return str.split("").reverse().join("");
 }
 
 export function checkPalindrome(str: string) {
-    let reversed = reverseAString(str);
-
-    str = str.replace(/\s+/g, '');
+    let normalizedStr = str.toLowerCase();
+    let reversed = reverseAString(normalizedStr);
+    str = normalizedStr.replace(/\s+/g, '');
     reversed = reversed.replace(/\s+/g, '');
-    return (str == reversed) ? true : false;
+    return (str == reversed);
```

Comment:
In this specific case, using == instead of === would not have any consequences because both operands (str and reversed) are of the same type (string).

However, as far as I know, it is generally considered good practice to use === instead of == to avoid unexpected behavior due to type coercion... Yes, it's unlikely to happen in TS because of type-checking, but it still could happen as any js code is a valid ts.

Reply:
Nope, you're correct, this isn't a case where I need to avoid strict equality, so there's really no reason not to use it.